galaxy_importer/loaders.py

```diff
@@ -80,44 +108,83 @@ def _validate_name(self):
     def _log_loading(self):
         self.log.info(' ')
         self.log.info(
-            f'===== LOADING {self.content_type.name}: {self.name} =====')
+            f'===== LOADING {self.content_type.name}: {self.path_name} =====')
 
 
 class PluginLoader(ContentLoader):
     def load(self):
         self._log_loading()
-        self.doc_strings = self._get_doc_strings()
+        doc_strings = DocStringLoader(
+            content_type=self.content_type.value,
+            path=self._get_tmp_dir(self.root),
+            fq_name=self._get_fq_name(self.root, self.path_name),
+            logger=self.log,
+        ).load()
 
         return schema.Content(
-            name=self.name,
+            name=self.path_name,
             content_type=self.content_type,
-            doc_strings=self.doc_strings,
+            doc_strings=doc_strings,
         )
 
-    def _make_name(self):
-        return os.path.splitext(os.path.basename(self.rel_path))[0]
+    @staticmethod
+    def _make_name(rel_path):
+        return os.path.splitext(os.path.basename(rel_path))[0]
+
+    @staticmethod
+    def _make_path_name(rel_path, name):
+        dirname_parts = Path(os.path.dirname(rel_path)).parts[2:]
+        return '.'.join(list(dirname_parts) + [name])
+
 
-    def _get_doc_strings(self):
-        if self.content_type.value not in ANSIBLE_DOC_SUPPORTED_TYPES:
+class DocStringLoader():
+    def __init__(self, content_type, path, fq_name, logger=None):
+        self.content_type = content_type
+        self.path = path
+        self.fq_name = fq_name
+        self.log = logger or default_logger
+
+    def load(self):
+        if self.content_type not in ANSIBLE_DOC_SUPPORTED_TYPES:
             return None
 
         self.log.info('Getting doc strings via ansible-doc')
         json_output = self._run_ansible_doc()
 
         if not json_output:
             return
-        data = json.loads(json_output)
-        if self.name not in data.keys():
-            self.log.error(f'No "{self.name}" key in ansible-doc output')
-            return None
 
+        data = json.loads(json_output)
+        if not isinstance(data, dict):
+            raise exc.ImporterError('ansible-doc output not dictionary as expected')
+        if len(data.keys()) != 1:
+            raise exc.ImporterError('ansible-doc output did not return single top-level key')
+        data = list(data.values())[0]
         data = self._transform_doc_strings(data)
+
         return {
-            key: data[self.name].get(key, None)
+            key: data.get(key, None)
             for key in ANSIBLE_DOC_KEYS
         }
 
-    def _transform_doc_strings(self, data):
+    def _run_ansible_doc(self):
```

Comment:
+1 for having the method that executes some script and returns it output separate by itself. ie, one method to run the external command and return it's output, and another to actually parse it. Tends to make it easier to test, since you can test the parsing methods with canned output without having to do too much mock gymnastics. Really helpful when the output of the ext commands start changing across versions and you need to test multiple expected outputs.